packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -921,9 +917,298 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
   }
 
   @override
-  RevealedOffset getOffsetToReveal(RenderObject target, double alignment, { Rect? rect }) {
-    // TODO(Piinks): Add this back in follow up change (ensureVisible), https://github.com/flutter/flutter/issues/126299
-    return const RevealedOffset(offset: 0.0, rect: Rect.zero);
+  RevealedOffset getOffsetToReveal(
+    RenderObject target,
+    double alignment, {
+    Rect? rect,
+    AxisDirection? axisDirection,
+  }) {
+    // We must know which axis we are revealing for, since Offset refers to only
+    // one of two scroll positions.
+    assert(axisDirection != null);
+
+    // Starting at `target` and walking towards the root:
+    //  - `child` will be the last object before we reach this viewport, and
+    //  - `pivot` will be the last RenderBox before we reach this viewport.
+    RenderObject child = target;
+    RenderBox? pivot;
+    // Find the direct child of the viewport that contains our target
+    while (child.parent != this) {
+      final RenderObject parent = child.parent!;
+      if (child is RenderBox) {
+        pivot = child;
+      }
+      child = parent;
+    }
+
+    final double offset = switch (axisDirectionToAxis(axisDirection!)) {
+      Axis.vertical => verticalOffset.pixels,
+      Axis.horizontal => horizontalOffset.pixels,
+    };
+
+    // `rect` in the new intermediate coordinate system.
+    final Rect rectLocal;
+    // Our new reference frame render object's main axis extent.
+    final double pivotExtent;
+    if (pivot != null) {
+      assert(pivot.parent != null);
+      assert(pivot.parent != this);
+      assert(pivot != this);
+      assert(pivot.parent is RenderBox);
```

Comment:
This contradicts the     `//  - `pivot` will be the last RenderBox before we reach this viewport.` comment above. But maybe this will be deleted since I don't think we need to find the `pivot` at all?

Reply:
Ah nvm it is still true since everyone's a RenderBox.

Reply:
I was able to rework this without pivot - thanks for pointing this out!